### ISSUE TITLE
Home Page not auto-redirecting to clerk auth page on localhost in "@clerk/nextjs": "^5.0.3"

### ISSUE THREAD

**Issue by svashish305** [clerk/javascript#3294]
### Preliminary Checks

- [X] I have reviewed the documentation: https://clerk.com/docs

- [X] I have searched for existing issues: https://github.com/clerk/javascript/issues

- [X] I have not already reached out to Clerk support via email or Discord (if you have, no need to open an issue here)

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Clerk. Please ask questions in our Discord community: https://clerk.com/discord.

### Reproduction

https://github.com/svashish305/sv_imaginify

### Publishable key

pk_test_Y2hlZXJmdWwtY2F0LTE4LmNsZXJrLmFjY291bnRzLmRldiQ

### Description

**Steps to reproduce:**

1. Step 1: Run the nextjs clerk app
2. Step 2: After following this [official docs guide](https://clerk.com/docs/references/nextjs/custom-signup-signin-pages) for my [repo](https://github.com/svashish305/sv_imaginify), navigate to localhost:3000 in chrome incognito browser.

**Expected behavior:**

Home page ('localhost:3000/') should auto-redirect to clerk auth page

**Actual behavior:**

There's no auto-redirect. If I manually navigate to localhost:3000/sign-in, I get the signin, same for signup, but there's no auto redirect setup.

### Environment

```Shell
System:
    OS: Windows 11 10.0.22621
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 3.44 GB / 15.81 GB
  Binaries:
    Node: 20.11.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.2.4 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (123.0.2420.97)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @clerk/nextjs: ^5.0.3 => 5.0.3
    @types/node: ^20 => 20.12.7
    @types/qs: ^6.9.15 => 6.9.15
    @types/react: ^18 => 18.3.1
    @types/react-dom: ^18 => 18.3.0
    class-variance-authority: ^0.7.0 => 0.7.0    
    clsx: ^2.1.1 => 2.1.1
    eslint: ^8 => 8.57.0
    eslint-config-next: 14.2.3 => 14.2.3
    lucide-react: ^0.376.0 => 0.376.0
    next: 14.2.3 => 14.2.3
    postcss: ^8 => 8.4.38
    qs: ^6.12.1 => 6.12.1
    react: ^18 => 18.3.1
    react-dom: ^18 => 18.3.1
    tailwind-merge: ^2.3.0 => 2.3.0
    tailwindcss: ^3.4.1 => 3.4.3
    tailwindcss-animate: ^1.0.7 => 1.0.7
    typescript: ^5 => 5.4.5
```

**Comment by panteliselef** [comment]
Hey @svashish305 after going through the repo you attached I didn't find why the behaviour you described would be the expected one.

If am missing something, please reply with the file where this logic exists in.

Note: `clerkMiddleware` requires you to write the logic yourself, example:

```ts
export default clerkMiddleware((auth) => auth().protect());
```

**Comment by svashish305** [comment]
@panteliselef I see
in one of the older clerk versions it was auto-redirecting from '/' to the sign-in page upon adding ClerkProvider, if this isn't expected in the latest version (5.0.3), then it makes sense to close this issue.

**Comment by Nazmul-Hassan10** [comment]
> @panteliselef I see in one of the older clerk versions it was auto-redirecting from '/' to the sign-in page upon adding ClerkProvider, if this isn't expected in the latest version (5.0.3), then it makes sense to close this issue.

Facing the same problems. 1st time using clerk and after the basic setup it's not redirecting to the sign in page. what should I do?

**Comment by panteliselef** [comment]
@Nazmul-Hassan10 Have you tried [this example ](https://clerk.com/docs/references/nextjs/clerk-middleware#protect-all-routes)from our documentation ?

**Comment by YuniqueUnic** [comment]
> @Nazmul-Hassan10 Have you tried [this example ](https://clerk.com/docs/references/nextjs/clerk-middleware#protect-all-routes)from our documentation ?

Thank you for your info, It's the solution to this issue.

I also encountered the same issue with no auto-redirect behavior on the HomePage. After numerous attempts and searches, I discovered the solution was right there in the documentation. ^_^…

![clerkMiddlewareForNextjs](https://github.com/user-attachments/assets/2470d852-6538-458d-bdff-e3399dc6b7dc)

As recorded in the [official docs for Next.js](https://clerk.com/docs/references/nextjs/clerk-middleware#protect-all-routes):

> By default, clerkMiddleware will not protect any routes. All routes are public and you must opt-in to protection for routes.

Therefore, it’s essential to define a rule to secure the routes that require authentication. If this is not done, the authentication prompt will not appear.
